File: Java/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left=0;
        int right=height.length-1;
        int max_area=0;
        while(left<right){
            int w=right-left;
            int h=Math.min(height[left],height[right]);
            int area= w * h;
            max_area = Math.max(max_area,area);
            if (height[left]<height[right])
                left++;
            else
                right--;
        }
        return max_area;
    }
}

File: Go/0011-container-with-most-water.go
func maxArea(height []int) int {
	left := 0
	right := len(height) - 1
	maxArea := 0
	for left < right {
		w := right - left
		h := min(height[left], height[right])
		area := w * h
		maxArea = max(maxArea, area)
		if height[left] < height[right] {
			left++
		} else {
			right--
		}
	}
	return maxArea
}